src/languages/python.js

```diff
@@ -194,12 +194,33 @@ export default function(hljs) {
     ]
   };
 
+  // https://docs.python.org/3.9/reference/lexical_analysis.html#numeric-literals
+  const digitpart = '[0-9](_?[0-9])*';
+  const pointfloat = `(${digitpart})?\\.(${digitpart})|(${digitpart})\\.`;
   const NUMBER = {
     className: 'number', relevance: 0,
     variants: [
-      {begin: hljs.BINARY_NUMBER_RE + '[lLjJ]?'},
-      {begin: '\\b(0o[0-7]+)[lLjJ]?'},
-      {begin: hljs.C_NUMBER_RE + '[lLjJ]?'}
+      // exponentfloat, pointfloat
+      // https://docs.python.org/3.9/reference/lexical_analysis.html#floating-point-literals
+      // optionally imaginary
+      // https://docs.python.org/3.9/reference/lexical_analysis.html#imaginary-literals
+      {begin: `((${digitpart})|(${pointfloat}))[eE][+-]?(${digitpart})[jJ]?`},
+      {begin: `(${pointfloat})[jJ]?`},
```

Comment:
I there a reason we're not using \b here as well?

Reply:
We can't use it there because floats can start or end with a decimal point, but I did incorporate it into the `pointfloat` pattern component and add explanatory comments.